src/@types/InteractionContext.d.ts

```diff
@@ -228,6 +229,10 @@ declare global {
 	interface ButtonContext extends InteractionContext {
 		component: APIButtonComponent | ButtonComponent;
 		message: Message;
```

Comment:
Now that I see this again, we should probably prevent assigning these properties, too.

```suggestion
		readonly component: APIButtonComponent | ButtonComponent;
		readonly message: Message;
```

Reply:
done